src/Elastic.Apm/Model/Span.cs

```diff
@@ -298,38 +290,10 @@ public void End()
 
 				// Spans are sent only for sampled transactions so it's only worth capturing stack trace for sampled spans
 				// ReSharper disable once CompareOfFloatsByEqualityOperator
-				if (ConfigSnapshot.StackTraceLimit != 0 && ConfigSnapshot.SpanFramesMinDurationInMilliseconds != 0)
-				{
-					if (Duration >= ConfigSnapshot.SpanFramesMinDurationInMilliseconds
-						|| ConfigSnapshot.SpanFramesMinDurationInMilliseconds < 0)
-					{
-						if (_stackFrames == null)
-						{
-							StackFrame[] trace;
-							var stackTrace = new StackTrace(true);
-							try
-							{
-								// I saw EnhancedStackTrace throwing exceptions in some environments
-								// therefore we try-catch and fall back to a non-demystified call stack.
-								trace = new EnhancedStackTrace(stackTrace).GetFrames();
-							}
-							catch
-							{
-								trace = stackTrace.GetFrames();
-							}
-
-							StackTrace = StacktraceHelper.GenerateApmStackTrace(trace,
-								_logger,
-								ConfigSnapshot, _apmServerInfo, $"Span `{Name}'");
-						}
-						else
-						{
-							StackTrace = StacktraceHelper.GenerateApmStackTrace(_stackFrames,
-								_logger,
-								ConfigSnapshot, _apmServerInfo, $"Span `{Name}'");
-						}
-					}
-				}
+				if (ConfigSnapshot.StackTraceLimit != 0 && ConfigSnapshot.SpanFramesMinDurationInMilliseconds != 0 && RawStackTrace == null
```

Comment:
Worth enclosing this logic in a well named method somewhere e.g. `ConfigSnapshot.CaptureStackTrace(double duration)`, or a private method on `Span`?